pipelines/tekton/aiedge-e2e/aiedge-e2e.pipeline.yaml

```diff
@@ -0,0 +1,306 @@
+                  periodSeconds: 5
+                  successThreshold: 1
+                readinessProbe:
+                  failureThreshold: 8
+                  httpGet:
+                    path: /v2/models/$(params.model-name)/ready
+                    port: 8080
+                    scheme: HTTP
+                  periodSeconds: 5
+                  successThreshold: 1
+                ports:
+                - containerPort: 8080
+                resources: {}
+        status: {}
+        EOF
+        oc wait deployment -n $(params.target-namespace) $(params.model-name)-$(params.model-version) --for condition=Available=True --timeout=120s
+        oc wait pod -n $(params.target-namespace) -l app=$(params.model-name)-$(params.model-version) --for condition=Ready=True --timeout=120s
+    taskRef:
+      kind: ClusterTask
+      name: openshift-client
+    runAfter:
+      - build-container
+  - name: create-default-service
+    params:
+    - name: SCRIPT
+      value: oc expose deployment  $(params.model-name)-$(params.model-version) --port=8080
+        --target-port=8080 --selector='app=$(params.model-name)-$(params.model-version)'
+        --dry-run=client -o yaml |  oc apply -f -
+    - name: VERSION
+      value: latest
+    runAfter:
+    - test-container-deploy
+    taskRef:
+      kind: ClusterTask
+      name: openshift-client
+  - name: test-model-rest-svc
+    params:
+    - name: model-name
+      value: $(params.model-name)
+    - name: model-version
+      value: $(params.model-version)
+    - name: test-endpoint
+      value: $(params.test-endpoint)
+    runAfter:
+    - create-default-service
+    taskRef:
+      kind: Task
+      name: test-model-rest-svc
+    workspaces:
+    - name: test-data
+      workspace: test-data
+  - name: stop-deployment
+    params:
+    - name: SCRIPT
+      value: |
+        if [ "$(params.upon-end)" == "stop" ]; then
+          oc scale deployment.apps/$(params.model-name)-$(params.model-version) --replicas=0
+        elif [ "$(params.upon-end)" == "delete" ]; then
+          oc delete all --selector=app=$(params.model-name)-$(params.model-version)
+        elif [ "$(params.upon-end)" == "keep" ]; then
+          echo "Keeping the deployment running."
+        else
+          echo "Invalid value for upon-end parameter."
+          exit 1
+        fi
+    - name: VERSION
+      value: latest
+    runAfter:
+    - test-model-rest-svc
+    taskRef:
+      kind: ClusterTask
+      name: openshift-client
+  - name: retrieve-image-digest
+    taskRef:
+      kind: Task
+      name: retrieve-image-digest
+    params:
+    - name: model-name
+      value: $(params.model-name)
+    - name: model-version
+      value: $(params.model-version)
+    - name: namespace
+      value: $(params.target-namespace)
+    - name: target-registry-url
+      value: quay.io/$(params.target-imagerepo)/$(params.model-name)
+    runAfter:
+    - test-model-rest-svc
+    - stop-deployment
```

Comment:
Maybe remove this line since the deployment still running doesn't hinder uploading the image?

Reply:
I can not remember why I originally added `stop-deployment` beyond visually representing that `stop-deployment is not an end state in the Pipeline details.  As the pipeline evolves, `stop-deployment` functionality should merged into `test-model-rest-svc'